Extensions/BBText/JsExtension.js

```diff
@@ -0,0 +1,601 @@
+      objectProperties.set(
+        'fontSize',
+        new gd.PropertyDescriptor(objectContent.fontSize)
+          .setType('number')
+          .setLabel(_('Base size'))
+      );
+
+      objectProperties.set(
+        'align',
+        new gd.PropertyDescriptor(objectContent.align)
+          .setType('choice')
+          .addExtraInfo('left')
+          .addExtraInfo('center')
+          .addExtraInfo('right')
+          .setLabel(_('Base alignment'))
+      );
+
+      objectProperties.set(
+        'fontFamily',
+        new gd.PropertyDescriptor(objectContent.fontFamily)
+          .setType('string')
+          .setLabel(_('Base font family'))
+      );
+
+      objectProperties.set(
+        'wordWrap',
+        new gd.PropertyDescriptor(objectContent.wordWrap ? 'true' : 'false')
+          .setType('boolean')
+          .setLabel(_('Word wrapping'))
+      );
+
+      objectProperties.set(
+        'visible',
+        new gd.PropertyDescriptor(objectContent.visible ? 'true' : 'false')
+          .setType('boolean')
+          .setLabel(_('Visible on start'))
+      );
+
+      return objectProperties;
+    };
+    objectBBText.setRawJSONContent(
+      JSON.stringify({
+        text:
+          '[b]bold[/b] [i]italic[/i] [size=15]smaller[/size] [font=times]times[/font] font\n[spacing=12]spaced out[/spacing]\n[outline=yellow]outlined[/outline] [shadow=red]DropShadow[/shadow] ',
+        opacity: 255,
+        fontSize: '20',
+        visible: true,
+        color: '#000000',
+        fontFamily: 'Arial',
+        align: 'left',
+        wordWrap: true,
+      })
+    );
+
+    objectBBText.updateInitialInstanceProperty = function(
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      return false;
+    };
+    objectBBText.getInitialInstanceProperties = function(
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'BBText',
+        _('BBText'),
+        _(
+          'Displays a rich text label using BBCode markup (allowing to set parts of the text as bold, italic, use different colors and shadows).'
+        ),
+        'JsPlatform/Extensions/bbcode32.png',
+        objectBBText
+      )
+      .setIncludeFile('Extensions/BBText/bbtextruntimeobject.js')
+      .addIncludeFile('Extensions/BBText/bbtextruntimeobject-pixi-renderer.js')
+      .addIncludeFile('Extensions/BBText/dist/pixi-multistyle-text.umd.js');
+
+    // Utility function to add both a setter and a getter to a property from a list. Useful for setting multiple generic properties
+    const addSettersAndGettersToObjectHelper = (
+      gdObject,
+      properties,
+      objectName
+    ) => {
+      properties.forEach(property => {
+        property.type = property.type === 'boolean' ? 'yesorno' : property.type;
```

Comment:
Don't modify the arguments, it's counter intuitive and error prone for the caller to have the object passed as parameter changed (we say that the function has "side-effects").
Instead, simply use another variable:

```
const parameterType = property.type === 'boolean' ? 'yesorno' : property.type;
```

and use `parameterType` then. You can see that I chose a name that seems adapted to the fact that it's used in a parameter (because "yesorno" is a type of parameter)